### DISCUSSION TITLE
Elastic ingest pipeline for Palo Alto firewalls

### DISCUSSION THREAD

**Discussion by Keenenick** [Security-Onion-Solutions/securityonion#3849] [Unsupported Versions] [answered]
Good day,

I would like to ask if anyone has an ingest pipeline for palo alto firewalls they are willing to share. I am currently ingesting the raw logs but I am not sure how properly parse them to get the data in a format SO2 will read them.

Thanks for any help.

Nick

**Comment by pboosten** [comment]
Hi,

I think you'll have to write that yourself. Good thing is that most of the work already has been done by someone. If you download filebeat from elasticsearch it contains a module called panw, which holds a pipeline file in yaml format. This can easily be converted to json. PANW stands for Palo Alto Networks.

There are actually two things you need to change: the default ingest pipeline for syslog is called 'syslog': this pipeline is actually the one that tries to redirect to the 'real' pipeline. It doesn't do that by  default, and therefore needs to be overridden.

The second file  to change is that palo alto pipeline that comes with filebeat. Going through the grok processor in the syslog pipeline messes up the log event a bit.

If you drop your (modified) files into  
/opt/so/saltstack/local/salt/elasticsearch/files/ingest
and restart elastic (so-elasticsearch-restart) then your pipelines get imported into SO (if all goes well, of course).

The original syslog pipeline can be found in
/opt/so/saltstack/default/salt/elasticsearch/files/ingest

**you should not modify this file, but place a copy in the first directory**

Hope this helps.

Peter

**Reply by sinnetll19** [reply]
did you get the palo alto logs ingested correctly? if so, can you share detailed steps on how to get it accomplished. peter's answer is helpful however its not detailed enough for me to figure out how to do this. it will be extremely helpful in my environment to get the palo logs into so2 correctly. i appreciate any help you can provide. thank you

**Comment by abesinger** [comment, marked as answer]
Um, maybe I'm missing something, but I'm not sure why one would have to write their own pipeline or mess with anything like that.  I'm using the filebeat panw module.  I have Palo Alto hosts sending their logs to a sensor node running filebeat  with the panw module, which in turn sends the logs to SO/Elastic.   It's all done with the SO config.  The filebeat panw ingest pipeline gets automagically loaded in the process. No customization/coding necessary.

The logs get indexed in Elastic to `so-panw-*`

All I had to do was:

- Add the following to /opt/so/saltstack/local/pillar/minions/<node>_sensor.sls:
```filebeat:
    third_party_filebeat:
        modules:
            panw:
              panos:
                enabled: true
                var.syslog_host: 0.0.0.0
                var.syslog_port: 9002
```
- Configure the sensor firewall to accept udp on port 9002 from the Palo Alto boxen (see [The Fine Security Onion doc](https://docs.securityonion.net/en/2.3/firewall.html) for details
- Point the Palo Alto logger at port 9002 on the sensor node.

**Reply by sinnetll19** [reply]
which version of so2 comes with the filebeat panw module?

**Reply by abesinger** [reply]
I can't tell you the _oldest_ version, but I had it running on 2.3.70, and maybe 2.3.60. FWIW, filebeat modules are bundled with filebeat, so if you're running and older version of SO and want to see if you have the module, look at what version of filebeat is installed, and then go look at the Filebeat site for the that version.

**Reply by pboosten** [reply]
When I wrote this we were still on 2.3.20, and there was absolutely nothing in the Fine Security Onion doc, except for a referral to Elastic pipelines.

In the meantime things are improved, and I'm very glad about it.

**Reply by sinnetll19** [reply]
im running 2.3.70. it should have the panw module but it is not listed under /opt/so/conf/filebeat/modules/thirdparty.yml. i tried all the configs above and its still not working.  there doesnt seem to be any index so-panw-* . the syslog data is coming in but its not getting parsed correctly. i do see the panw ingest pipeline in stack management.

**Reply by sinnetll19** [reply]
i got it to work. i was making a simple mistake. but your solution does work 100%. thank you abesinger for your help. do you know where i have to add the above lines for panw to work on a standalone SO?

**Reply by abesinger** [reply]
Sorry, can't help with that one. I've only done a distributed install.

**Reply by ddbrandonb** [reply]
> i got it to work. i was making a simple mistake. but your solution does work 100%. thank you abesinger for your help. do you know where i have to add the above lines for panw to work on a standalone SO?

What exactly was your simple mistake if you don't mind me asking? I am in the same state. filebeat is sending the panos logs to logstash but no indexes are being created on any of my search nodes. I see the ingest pipeline there also. Any information is greatly appreciated!

**Reply by ddbrandonb** [reply]
I solved my issue. My manager /opt/so/saltstack/local/pillar/minions/*_searchnode.sls configs had some custom pipeline configs copied from an older SO version salt default. The "so/9050_output_filebeatmodules.conf.jinja" template was missing from them. All is good now -- thank you @abesinger !

**Reply by mnasec** [reply]
Worked perfect for me since 2.3.130 in a distributed installation with manger, forward and searchnoes, but not in 2.3.140 anymore, because i only enabled the panw module within the sensor minions, so the search nodes did not know about the panw pipelines. So i enabled the panw module in the global.sls also - see  #8481

**Reply by dougburks** [reply]
We've updated our Filebeat documentation to reflect the change:
https://docs.securityonion.net/en/2.3/filebeat.html
